pyro/nn/auto_reg_nn.py

```diff
@@ -52,16 +52,17 @@ def create_mask(input_dim, observed_dim, hidden_dims, permutation, output_dim_mu
 
     # Create the indices that are assigned to the neurons
     input_indices = torch.cat((torch.zeros(observed_dim), 1 + var_index))
-    hidden_indices = [sample_mask_indices(input_dim, h) for h in hidden_dims]
+    if observed_dim > 0:
+        hidden_indices = [sample_mask_indices(input_dim, h) for h in hidden_dims]
+    else:
+        hidden_indices = [sample_mask_indices(input_dim - 1, h) + 1 for h in hidden_dims]
```

Comment:
This change works perfectly! Thanks @stefanwebb !

Reply:
No worries!

Reply:
Can we reduce the offset of hidden_indices by `1` and change the inequality at lines 66 and 71 to `>=` and `>` instead? There is no harm with it, I just want to make it clear that they are indices as in the figure in MADE paper. :)

Reply:
Okay, I've made this change!